### PR TITLE
Add focus functionality for devices without touch screen

### PR DESCRIPTION
**Pull request by alimyhamed**
(faced problem while developing for android TV, cause there only remote controller for device)

### CODE REVIEW

**Comment by Titozzz** [comment]
Are we sure there is no side effect, if you load a webview next to a form, or anything else? I'm not confident with the approach and would rather have a focus() method you could call on the ref

**Comment by alimyhamed** [comment]
> This looks good to me. Have you tested it on the device in question, @alimyhamed?

Yes, I've tested it on tv device, this is the only option that helped me, at least trigger focus on input.

**Comment by alimyhamed** [comment]
> Are we sure there is no side effect, if you load a webview next to a form, or anything else? I'm not confident with the approach and would rather have a focus() method you could call on the ref

It would be great, but its the only option that I found to fix the issue. 
For more, I don't have enough knowledge.
In our app, we need to open a bank payment form and focus on the input to open the keyboard and start interacting with a form. And it's pretty hard if you don't have a possibility to focus or click.

**Comment by alimyhamed** [comment]
Thanks for the help!

**Comment by Titozzz** [comment]
I sadly don't have much time right now, I'll tag this as needing help!

**Comment by jamonholmgren** [comment]
@Titozzz I think this is worth implementing as a small change. If we get reported problems, we can roll it back easily (1-line) and implement as a ref method like you suggest.

**Comment by jamonholmgren** [comment]
After discussion with @Titozzz , we think this will cause problems when released.

As an alternative, @alimyhamed, can you implement as a ref method? You can see examples of how to do that by searching the code base for `goBack`.

https://github.com/react-native-community/react-native-webview/search?q=goBack

Try looking over how that's implemented and see if you can take a crack at it. Make sure to updated documentation too.

File: android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java [hunk 1]
```diff
@@ -470,6 +472,8 @@ public Map getExportedCustomDirectEventTypeConstants() {
       "injectJavaScript", COMMAND_INJECT_JAVASCRIPT,
       "loadUrl", COMMAND_LOAD_URL
     );
+    map.put("requestFocus", COMMAND_FOCUS);
```
Comment (Titozzz) [inline android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java]: You don't need to store mapbuilder to a variable, you can just add it above

File: android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java [hunk 2]
```diff
@@ -379,6 +380,7 @@ public void setSource(WebView view, @Nullable ReadableMap source) {
           }
         }
         view.loadUrl(url, headerMap);
+        view.requestFocus();
```
Comment (Titozzz) [inline android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java]: Remove this so focus only happens when user calls the function

**Comment by Titozzz** [comment]
Thanks for taking the effort. Just a few fixes needed

File: android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java (hunk 2 above)
Comment (alimyhamed) [reply android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java]: Forget about this line.

File: android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java (hunk 1 above)
Comment (alimyhamed) [reply android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java]: Sorry, I'm not sure how.
Cause MapBuiler is 7 items max.
image

File: android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java (hunk 2 above)
Comment (alimyhamed) [reply android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java]: https://github.com/react-native-community/react-native-webview/pull/567/commits/5c31cf3b12ba94569b17bd2fdabb2d54518d8b86

File: android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java (hunk 1 above)
Comment (alimyhamed) [reply android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java]: Can someone help with mapbuilder fix?

**Comment by WarrenMoore** [comment]
Could I ask what's left fix-wise on this before you'd be able to consider merging it in? I see your concern @Titozzz is with how `MapBuilder.of` isn't being used but as @alimyhamed pointed out overrides only exist for a maximum of 7 pairings (https://github.com/facebook/react-native/blob/master/ReactAndroid/src/main/java/com/facebook/react/common/MapBuilder.java#L84).

**Pull request merged**